tooling/cli/src/interface/rust.rs

```diff
@@ -255,6 +257,59 @@ impl Interface for Rust {
   }
 }
 
+struct IgnoreMatcher(Vec<Gitignore>);
+
+impl IgnoreMatcher {
+  fn is_ignore(&self, path: &Path, is_dir: bool) -> bool {
+    for gitignore in &self.0 {
+      if gitignore.matched(path, is_dir).is_ignore() {
+        return true;
+      }
+    }
+    false
+  }
+}
+
+fn build_ignore_matcher(dir: &Path) -> IgnoreMatcher {
+  let mut matchers = Vec::new();
+
+  // ignore crate doesn't expose an API to build `ignore::gitignore::GitIgnore`
+  // with custom ignore file names so we have to walk the directory and collect
+  // our custom ignore files and add it using `ignore::gitignore::GitIgnoreBuilder::add`
+  for entry in ignore::WalkBuilder::new(dir)
+    .require_git(false)
+    .ignore(false)
+    .overrides(
+      ignore::overrides::OverrideBuilder::new(dir)
+        .add(".taurignore")
+        .unwrap()
+        .build()
+        .unwrap(),
+    )
+    .build()
+    .flatten()
+  {
+    let path = entry.path();
+    if path.file_name() == Some(OsStr::new(".taurignore")) {
+      let mut ignore_builder = GitignoreBuilder::new(path.parent().unwrap());
+
+      ignore_builder.add(path);
+
+      if let Ok(ignore_file) = std::env::var("TAURI_DEV_WATCHER_IGNORE_FILE") {
+        ignore_builder.add(dir.join(ignore_file));
+      }
+
+      for line in crate::dev::TAURI_DEV_WATCHER_GITIGNORE.lines().flatten() {
+        let _ = ignore_builder.add_line(None, &line);
+      }
+
+      matchers.push(ignore_builder.build().unwrap());
+    }
+  }
+
+  IgnoreMatcher(matchers)
+}
+
```

Comment:
I am not sure why you need to store multiple ignore matcher, we just need to collect all `.taurignore` files in the common parent and add it to a single ignore matcher
```suggestion
fn build_ignore_matcher(dir: &Path) -> GitIgnore {
  let mut ignore_builder = GitignoreBuilder::new(path.parent().unwrap());

  // ignore crate doesn't expose an API to build `ignore::gitignore::GitIgnore`
  // with custom ignore file names so we have to walk the directory and collect
  // our custom ignore files and add it using `ignore::gitignore::GitIgnoreBuilder::add`
    for entry in ignore::WalkBuilder::new(dir)
      .require_git(false)
      .ignore(false)
      .overrides(
        ignore::overrides::OverrideBuilder::new(dir)
          .add(".taurignore")
          .unwrap()
          .build()
          .unwrap(),
      )
      .build()
      .flatten()
  {
    let path = entry.path();
    ignore_builder.add(path);

    if let Ok(ignore_file) = std::env::var("TAURI_DEV_WATCHER_IGNORE_FILE") {
      ignore_builder.add(dir.join(ignore_file));
    }

    for line in crate::dev::TAURI_DEV_WATCHER_GITIGNORE.lines().flatten() {
      let _ = ignore_builder.add_line(None, &line);
    }
  }

	ignore_builder
}

```

Reply:
well I tried that, but then relative paths doesn't work. For instance, if you have a workspace with the `lib/` and `src-tauri/` members, and a `lib/.taurignore` file ignoring `src/lib.rs` (which relatively to the common ancestor is `lib/src/lib.rs`), it only ignores properly if you use a dedicated `GitignoreBuilder::new(path)` where `path` points to the `lib/` folder.

Reply:
See [GitignoreBuilder::new](https://docs.rs/ignore/latest/ignore/gitignore/struct.GitignoreBuilder.html#method.new) for more info, particularly:
```
The path given should be the path at which the globs for this gitignore file should be matched. Note that paths are always matched relative to the root path given here. Generally, the root path should correspond to the directory containing a .gitignore file.
```

Reply:
Ah, I think I only tested with `.taurignore` in the root. This is fine then.